android-components/components/support/utils/src/main/java/mozilla/components/support/utils/ext/Fragment.kt

```diff
@@ -0,0 +1,25 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.support.utils.ext
+
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.fragment.app.Fragment
+
+/**
+ * Requests permissions and handles results inside a [Fragment].
+ */
+fun Fragment.requestInPlacePermissions(
+    permissionsToRequest: Array<String>,
+    onResult: (Map<String, Boolean>) -> Unit,
+) {
+    requireActivity().activityResultRegistry.register(
+        "permissionsRequest",
```

Comment:
I'm not that familiar with the new API, I'm a bit confused about having the [key](https://developer.android.com/reference/androidx/activity/result/ActivityResultRegistry#register(java.lang.String,androidx.lifecycle.LifecycleOwner,androidx.activity.result.contract.ActivityResultContract%3CI,O%3E,androidx.activity.result.ActivityResultCallback%3CO%3E)) parameter as a fixed string ""permissionsRequest". Could you help me to clarify why ? Thanks in advance :)

Reply:
Before we has request code, as an identifier for each request now it seems that we has the same identifier fall of the requests?

Reply:
Is it possible to add tests for this extensions function?

Reply:
It's not a new API, it has been used in Focus for a while.

Reply:
Thank you for your review, I will look into this more.

Reply:
As far as I have seen in tests, the key parameter in a `register` call is used to avoid mid-air collisions by requesting the same permission multiple times. This should not happen anyway, but we might send unique keys depending on the caller. maybe class name of the feature, etc